src/physics/utils/phys_comp.F90

```diff
@@ -177,12 +177,13 @@ subroutine phys_init()
       use cam_thermo_formula,   only: cam_thermo_formula_init
       use physics_types,        only: allocate_physics_types_fields
       use cam_ccpp_cap,         only: cam_ccpp_physics_initialize
+      use cam_constituents,     only: num_advected
 
       call cam_thermo_init(columns_on_task, pver, pverp)
       call cam_thermo_formula_init()
 
       call allocate_physics_types_fields(columns_on_task, pver, pverp,        &
-           set_init_val_in=.true., reallocate_in=.false.)
+           num_advected, set_init_val_in=.true., reallocate_in=.false.)
```

Comment:
Can I ask if this change is still necessary? Since `physics_types` is generated I wasn't sure if a framework change is necessary to accompany this change.

Reply:
I believe this code change was given to me by @peverwhee or @nusbaume.  I tried taking it out and I get the following compilation error: `Error: Missing actual argument for argument 'number_of_ccpp_constituents' at (1)`, so I deem it is necessary.  One of them can probably provide an explanation.

Reply:
hiya! `physics_types` is generated from the registry by CAM-SIMA rather than the CCPP framework. The new argument was added because Cheryl introduced a new dimension in the registry (number_of_ccpp_constituents / num_advected), which caused the generated code to expect this additional input

Reply:
Thanks @cacraigucar and @peverwhee for explaining!